pyanaconda/kickstart.py

```diff
@@ -2162,12 +2164,39 @@ def parseKickstart(f):
     # only checks if they need formatting, which requires zerombr to be known
 
     try:
-        ksparser.readKickstart(f)
+        # Process warnings differently in this part.
+        with warnings.catch_warnings():
+
+            # Treat the warnings as errors in the strict mode,
+            # otherwise just print them to logs.
+            action = "error" if strict_mode else "always"
+
+            # Set up the filter for pykickstart warnings.
+            for category in (UserWarning, SyntaxWarning, DeprecationWarning):
+                warnings.filterwarnings(action=action, category=category, module="pykickstart")
+
+            # Parse the kickstart file.
+            ksparser.readKickstart(f)
+
+    # We do not have an interface here yet, so we cannot use our error
+    # handling callback.
     except KickstartError as e:
-        # We do not have an interface here yet, so we cannot use our error
-        # handling callback.
-        print(e)
+        log.error(e)
+        msg = _("\nAn error occurred during reading the kickstart file:"
+                "\n%s\n\nThe installer will now terminate.") % str(e).strip()
+        print(msg)
+        iutil.ipmi_report(IPMI_ABORTED)
+        time.sleep(10)
+        sys.exit(1)
+
+    except (UserWarning, SyntaxWarning, DeprecationWarning) as w:
+        log.error(w)
+        msg = _("\nA warning occurred during reading the kickstart file:"
+                "\n%s\n\nPlease modify your kickstart file or remove the "
+                "`strict` option.\nThe installer will now terminate.") % str(w).strip()
+        print(msg)
```

Comment:
If I understand this correctly, it will only report the first error in the kickstart file. Or am I wrong? Is there a way to report all errors at once? Running the installation again just to find the next errors sounds like a lot of pain.